brave/brave-core
# (Wallet) Convert "Add custom asset" dialog to full screen

## Resolves https://github.com/brave/brave-browser/issues/27070

 This PR implements a full screen activity to add a custom asset or an NFT.
It also fixes some minor issues to make it consistant with iOS and Desktop implementation:
* For Solana tokens the label `Token address` becomes `Mint address`.
* The new UI got rid of the cancel button as the same action can be achieved by tapping on the back arrow icon or using the system navigation controls.
* Token ID for ERC-721 is not shown for Solana NFTs.
* Text watcher has been improved moving its implementation into a class interface.

Minor improvements to `NetworkSpinnerAdapter` class.

https://github.com/brave/brave-core/assets/3308503/f9d904a9-ee23-4f06-ab12-d205b2ae8fc0


## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### Add custom asset test plan

* Open Brave Wallet
* In the portfolio section scroll down and tap on "Edit visible assets"
* Tap on "Add custom asset"
* Add a new custom asset e.g. Alchemix (https://coinmarketcap.com/currencies/alchemix/)
* Tap "Add"
* Observe the new asset is shown in the list and is checked
* Tap "Save account watchlist"
* Observe the new asset is shown in the main portfolio list

### Add NFT test plan

* Open Brave Wallet
* In the NFTs section scroll down and tap on "Edit visible assets"
* Tap on "Add NFT"
* Add a new NFT e.g. Insivible Friends (https://opensea.io/assets/ethereum/0x59468516a8259058bad1ca5f8f4bff190d30e066/183)
* Tap "Add"
* Observe the new NFT is shown in the list and is checked
* Tap "Save account watchlist"
* Observe the new NFT is shown in the main portfolio list
